clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/validators/StandardAppEngineAttributeValidator.groovy

```diff
@@ -233,4 +236,32 @@ class StandardAppEngineAttributeValidator {
     }
     return valid
   }
+
+  def validateServingStatusCanBeChanged(String serverGroupName,
+                                        AppEngineNamedAccountCredentials credentials,
+                                        AppEngineClusterProvider clusterProvider,
+                                        String attribute) {
+    def valid = true
+    def serverGroup = clusterProvider.getServerGroup(credentials.name, credentials.region, serverGroupName)
+    if (!serverGroup) {
+      errors.rejectValue("${context}.${attribute}",
+                         "${context}.${attribute}.invalid (Server group $serverGroupName not found).")
+      valid = false
+      return valid
+    }
+
+    def isFlex = serverGroup.env == AppEngineServerGroup.Environment.FLEXIBLE
+    def usesManualScaling = serverGroup.scalingPolicy?.type == ScalingPolicyType.MANUAL
+    def usesBasicScaling = serverGroup.scalingPolicy?.type == ScalingPolicyType.BASIC
+
+    if (!(isFlex || usesManualScaling || usesBasicScaling)) {
+      errors.rejectValue("${context}.${attribute}",
+                         "${context}.${attribute}.invalid (Only server groups that use the flexible environment," +
+                         " or use basic or manual scaling can be started or stopped).")
```

Comment:
Just a nit, but could make the order of the declarations and conditional clauses match the order used in the error message.